docs/status-list.md

```diff
@@ -0,0 +1,174 @@
+# Status-List Module
+
+Verifiable Credentials (VCs) can have a `credentialStatus` property.
+The [status-list module](../src/modules/status-list/) provides an API to enable the credential status using the [StatusList2021 specification](https://w3c-ccg.github.io/vc-status-list-2021/).
+
+A key feature of the implementation is the use of the EnergyWeb ENS (Ethereum Name Service) Role Definitions to authorize updates to the status lists. 
+
+## Sequences
+
+### Issue credential
+
+```mermaid
+sequenceDiagram
+participant ri as Role Issuer
+participant shsl as StatusList Module
+
+ri->>shsl: POST /status-list/entries with credential
+shsl->>shsl: verify issuer authorization
+shsl->>shsl: NamespaceStatusLists.createEntry
+shsl->>shsl: CredentialWithStatus.associateEntry
+shsl-->>ri: return credential with credentialStatus  
+
+ri->>ri: issue a VC
+```
+
+### Revoke a credential
+```mermaid
+sequenceDiagram
+participant rv as Revoker
+participant shsl as SSI-Hub StatusList Module
+
+rv->>shsl: request to revoke a credential (provide entry)
+shsl-->>rv: return status list credential to be signed 
+rv->>rv: sign credential
+rv->>shsl: return status list VC
+shsl->>shsl: check the DID/credentials of the revoker
+opt revoker is authorized 
+shsl->>shsl: persist updated status list
+end
+```
+
+### Verify a credential
+```mermaid
+sequenceDiagram
+participant vr as Verifier
+participant shsl as SSI-Hub StatusList Module
+
+vr->>shsl: request a status list for presented VC
+shsl-->>vr: return status list VC
+vr->>vr: check that status list proof is valid
+vr->>vr: check if presented VC is revoked
+```
+
+## Key Design Challenges and Solutions
+
+### Revocation Authorization is based on Role Definition
+It only makes sense for a revoker to sign a list that they are authorized to sign.
+This authorization is determined by the role definition of the role credential that a revoker is attempting to revoke.
+
+This means that it is not feasible have just one list in SSI Hub, for example.
+This is because it is not feasible that a given revoker is authorized to revoke all roles.
+
+**Chosen solution**:
+
+This authorization challenge is eliminated by having one list per role definition/namespace.
```

Comment:
Isn't status list created for each credential? https://github.com/energywebfoundation/ssi-hub/blob/d1b42b2f434e8e2251c3d1a2b875bde4f317c9af/src/modules/status-list/entities/namespace-status-lists.entity.ts#L49

Reply:
Good point, I changed the wording to be "having the credentials on each given list be only for one role."